cli/proc_state.rs

```diff
@@ -637,18 +649,26 @@ impl ProcState {
       }
     }
 
-    // FIXME(bartlomieju): hacky way to provide compatibility with repl
+    // FIXME(bartlomieju): this is a hacky way to provide compatibility with REPL
+    // and `Deno.core.evalContext` API. Ideally we should always have a referrer filled
+    // but sadly that's not the case due to missing APIs in V8.
     let referrer = if referrer.is_empty() && self.flags.repl {
-      deno_core::DUMMY_SPECIFIER
+      deno_core::resolve_url_or_path("./$deno$repl.ts").unwrap()
     } else {
-      referrer
+      deno_core::resolve_url_or_path(referrer).unwrap()
     };
-    if let Some(import_map) = &self.maybe_import_map {
-      import_map
-        .resolve(specifier, referrer)
-        .map_err(|err| err.into())
+
+    let maybe_resolver: Option<&dyn deno_graph::source::Resolver> =
+      if let Some(resolver) = &self.maybe_resolver {
+        Some(resolver.as_ref())
+      } else {
+        None
+      };
+    if let Some(resolver) = &maybe_resolver {
```

Comment:
It's not wholly clear to me why it needs to be converted to `&dyn Resolver` first.

Reply:
I'm not sure what you mean by converting - but we're using `dyn Resolver` to have a single interface from `deno_graph`; that can be used in `prepare_module_load` method and `resolve` methods on the `ProcState`.